### PR TITLE
Add example encoding for godoc ssz

### PR DESCRIPTION
**Pull request by prestonvanloon**
I wasn't satisfied with the docs on how to ssz.Encode something so here's an example.

### CODE REVIEW

File: shared/ssz/example_encode_test.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+		Field1: 10,
+		Field2: []byte{1, 2, 3, 4},
+	}
+
+	// We use a bytes.Buffer as our io.Writer.
+	buffer := new(bytes.Buffer)
+	// ssz.Encode writes the encoded data to the buffer.
+	if err := ssz.Encode(buffer, d); err != nil {
+		// There was some failure with encoding SSZ.
+		panic(err)
```
Comment (nisdas) [inline shared/ssz/example_encode_test.go]: Should it be a panic ? A normal error would be fine here

Comment (prestonvanloon) [reply shared/ssz/example_encode_test.go]: It's just for example, see below

image

Comment (prestonvanloon) [reply shared/ssz/example_encode_test.go]: I'll add another comment that the user should handle the error rather than panic.

Comment (nisdas) [reply shared/ssz/example_encode_test.go]: Ah k got it. Sounds good

**Pull request merged**